### PR TITLE
test: 3 new tests for SelectCoins function

### PR DESCRIPTION
**Pull request by akankshakashyap**
Three new tests have been added.

1. More coins should be selected when effective fee < long term fee.
2. Less coin should be selected when effective fee > long term fee.
3. If a coin is preselected, it should be selected even if disadvantageous.

### CODE REVIEW

**Comment by akankshakashyap** [comment]
@brunoerg @MarcoFalke @achow101 could you please review it, as it would be helpful for my summer of bitcoin proposal.

File: src/wallet/test/coinselector_tests.cpp [hunk 1]
```diff
@@ -344,6 +344,73 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
...
+        LOCK(wallet->cs_wallet);
+        wallet->SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+        wallet->SetupDescriptorScriptPubKeyMans();
+
+        std::vector<COutput> coins;
+
+        expected_result.Clear();
+        add_coin(coins, *wallet, 10 * CENT, 6 * 24, false, 0, true);
+        add_coin(coins, *wallet, 9 * CENT, 6 * 24, false, 0, true);
+        add_coin(coins, *wallet, 1 * CENT, 6 * 24, false, 0, true);
```
Comment (achow101) [inline src/wallet/test/coinselector_tests.cpp:381]: In 5c15850ded2a3195fae0dd238e1aebaaa46b13ce "3 new tests for SelectCoins function"

All of this setup is the same in all three test cases. These can be deduplicated and just have a single scope for all these cases.

Comment (akankshakashyap) [reply src/wallet/test/coinselector_tests.cpp]: made these changes.

**Comment by MarcoFalke** [comment]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by fanquake** [comment]
Given this hasn't been re-reviewed yet, I'd suggest improving your commit message. It should follow a similar form to the PR title / description, a title, followed by a newline, and then a description of the change. Irrelevant things like "removed duplicate setup from tests", should be dropped. i.e:
```bash
test: add 3 new test cases for SelectCoins()

1. More coins should be selected when effective fee < long term fee.
2. Less coin should be selected when effective fee > long term fee.
3. If a coin is preselected, it should be selected even if disadvantageous.
```

**Comment by achow101** [comment]
ACK 3f8def51d53a078a5ee71ec675b5e06b784147de

**Review by brunoerg — APPROVED** [review]
ACK 3f8def51d53a078a5ee71ec675b5e06b784147de

Nice adds!

**Pull request merged**